spond/spond.py

```diff
@@ -380,3 +403,47 @@ async def update_event(self, uid, updates: dict):
         async with self.clientsession.post(url, json=data, headers=headers) as r:
             self.events_update = await r.json()
             return self.events
+
+    async def update_event_response(
+        self,
+        uid,
+        accepted: bool = True,
+    ) -> List[dict]:
+        """
+        Respond to an event invite.
+
+        Parameters:
+        ----------
+        uid : str
+           UID of the event.
+        accepted : bool
+            Accept invite to an event.
+
+        Returns:
+        ----------
+        json results of put command, containing:
+            `acceptedIds`,
+            `unconfirmedIds`,
+            `waitinglistIds`,
+            `votes`
+        """
+        if not self.token:
+            await self.login()
+        profile = await self.get_profile()
+        event = await self.get_event(uid)
+
+        membershipId = None
+        members = event["recipients"]["group"]["members"]
```

Comment:
Is a similar test also needed in other functions?
Seems like it might be a good idea maybe?
And if so - it should probably be abstracted out to a separate function?

What do you think @elliot-100 ?

Reply:
Raise exception if the current user isn't a member of the event's group?

It makes sense that you can't respond to an event if you're not in its group - the event isn't available for you to attend. It's needed because you may or may not be able to read it depending on visibility settings.

I guess there are other 'write' cases where this should be checked but I'm not really familiar with them...

--

I do think the function name `get_profile()` should be `get_current_profile()` or similar, as the former sounds like a general purpose function to read an arbitrary item, like `get_event()`, get_group`, etc.